File: chapter-01/many-languages/Logger.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <chrono>
#include <iomanip>

int main() {
    std::ofstream output_file("application.log", std::ios::app);
    if (!output_file.is_open()) {
        std::cerr << "Error: Could not open log file application.log\n";
        return 1;
    }

    auto now = std::chrono::system_clock::now();
    auto in_time_t = std::chrono::system_clock::to_time_t(now);

    output_file << "[" << std::put_time(std::localtime(&in_time_t), "%Y-%m-%d %H:%M:%S") << "] INFO: Operation completed successfully." << std::endl;
    output_file.close();
    std::cout << "Logged message to application.log\n";
    return 0;
}

File: chapter-01/many-languages/Logger.js
const fs = require('fs');

const fileName = 'application.log';
const logMessage = 'INFO: Operation completed successfully.';

const now = new Date();
const timestamp = now.getFullYear() + '-' +
                  String(now.getMonth() + 1).padStart(2, '0') + '-' +
                  String(now.getDate()).padStart(2, '0') + ' ' +
                  String(now.getHours()).padStart(2, '0') + ':' +
                  String(now.getMinutes()).padStart(2, '0') + ':' +
                  String(now.getSeconds()).padStart(2, '0');

const logEntry = `[${timestamp}] ${logMessage}\n`;

fs.appendFile(fileName, logEntry, (err) => {
    if (err) {
        console.error('Error writing to log file:', err);
    } else {
        console.log(`Logged message to ${fileName}`);
    }
});